timesketch/frontend/src/components/Sketch/TimeFilter.vue

```diff
@@ -131,35 +237,51 @@ export default {
         [this.startDateTime, this.endDateTime] = [this.endDateTime, this.startDateTime]
       }
 
-      this.chip = {
+      let chip_type = ''
+      let chip_value = ''
+
+      // Set the right chip type and value
+      if (this.radio == 'interval') {
+        chip_type = 'datetime_interval'
+        chip_value = this.getOffsetDateTime()
+      }
+      else {
+        chip_type = 'datetime_range'
+        chip_value = this.startDateTime + ',' + this.endDateTime
+      }
+
+      // Update or creating a chip
+      if (this.chip) {
+        this.chip['type'] = chip_type
+        this.chip['value'] = chip_value
+        this.$emit('updateChip', this.chip)
+      }
+      else {
+        this.chip = {
           'field': '',
-          'value': this.startDateTime + ',' + this.endDateTime,
-          'type': 'datetime_range',
+          'type': chip_type,
+          'value': chip_value,
           'operator': 'must',
           'active' : true
         }
-      this.$emit('addChip', this.chip)
-      this.startDateTime = ''
-      this.endDateTime = ''
+        this.$emit('addChip', this.chip)
 
-      // Close the menu
-      this.$emit('hideDropdown')
-    },
-    update: function() {
-      if (!(this.startDateTime && this.endDateTime)) {
-        return
+        // Wipe out the values in the Create drop down
+        this.resetInterface()
       }
 
-      // The filter doesn't work if the start date is after the end date
-      if (this.startDateTime > this.endDateTime) {
-        [this.startDateTime, this.endDateTime] = [this.endDateTime, this.startDateTime]
-      }
-
-      this.chip['value'] = this.startDateTime + ',' + this.endDateTime;
-      this.$emit('updateChip', this.chip)
-
       // Close the menu
       this.$emit('hideDropdown')
+    },
+    resetInterface: function() {
+      Object.assign(this.$data, this.$options.data()) // Credits to https://stackoverflow.com/a/38174780
```

Comment:
Do you need the context here as well, seeing you are using `this`?
```
Object.assign(this.$data, this.$options.data.apply(this))
```

Reply:
I'm not sure how `this.$options.data.apply(this)` is different to `this.$options.data()`? The idea is only to reset the object's state to that when it was first initiated. Since the way you suggested works to reset the component, I've updated the code to what you suggested.